File: BackEnd/Projeto-Estacionamento/Entities/Carro.cs
namespace Projeto_Estacionamento.Entities
{
    public interface Carro
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Ano { get; set; }
        public int QuantidadeDePortas {  get; set; }
        public string ModeloDoMotor { get; set; }
        public string Placa { get; set; }
        public string Cor { get; set; }
    }
}

File: FrontEnd/GerenciadorDeEstacionamento/src/models/Carro.ts
import type { Marca } from "./Marca";
import type { TipoCarro } from "./TipoCarro";

export class Carro{
    id:number;
    modelo:string;
    ano:string;
    quantidadeDePortas:number;
    modeloDoMotor:string;
    placa:string;
    cor:string;
    tipoCarro:TipoCarro;
    marca:Marca;

    constructor(id:number, modelo:string, ano:string, quantidadeDePortas:number, modeloDoMotor:string, placa:string, cor:string, tipoCarro:TipoCarro, marca:Marca){
        this.id = id;
        this.modelo = modelo;
        this.ano = ano;
        this.quantidadeDePortas = quantidadeDePortas;
        this.modeloDoMotor = modeloDoMotor;
        this.placa = placa;
        this.cor = cor;
        this.tipoCarro = tipoCarro;
        this.marca = marca;
    }
}